x-pack/plugin/search-business-rules/src/yamlRestTest/resources/rest-api-spec/test/search-business-rules/10_pinned_retriever_basic.yml

```
@@ -0,0 +1,393 @@
+setup:
+  - requires:
+      cluster_features: 'pinned_retriever_supported'
+      reason: 'test requires pinned retriever implementation'
+  - do:
+      indices.create:
+        index: test-index1
+
+  - do:
+      bulk:
+        refresh: true
+        index: test-index1
+        body:
+          - index:
+              _id: doc1
+          - { "text": "document one" }
+          - index:
+              _id: doc2
+          - { "text": "document two" }
+          - index:
+              _id: doc3
+          - { "text": "document three" }
+          - index:
+              _id: doc4
+          - { "text": "document four" }
+          - index:
+              _id: doc5
+          - { "text": "document five" }
+
+---
+"pinned retriever parameter variations":
+  - skip: { features: headers }
+  - do:
+      headers: { Content-Type: application/json }
+      search:
+        index: test-index1
+        body:
+          retriever:
+            pinned:
+              docs:
+                - _index: test-index1
```

can we get a test with a doc id which does not exist?
to make sure we don't have any assumptions in our implementation that we a document with the given ID always exists?

---

Thanks for the suggestion I’ve added a test with a non-existent doc ID to verify the behaviour.

Right now, the retriever silently ignores missing IDs, similar to how terms or ids queries behave in Elasticsearch. This favours resilience by pinning valid docs and skipping over invalid ones without failing the search.

That said, if we’d prefer stricter validation (e.g., erroring on missing IDs), we can explore that as a design choice possibly as an optional setting.

Let me know if you’re good with the current behaviour, or if you’d prefer we make it stricter.